### PR TITLE
feat: add support for Devanagari & Bengali decimal numbering systems

### PR DESCRIPTION
**Pull request by Rocss**
Closes  https://github.com/adobe/react-spectrum/issues/6623

## ✅ Pull Request Checklist:

- [x] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [x] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/WAI/ARIA/apg/)

## 🧢 Your Project:

Adobe Express

### CODE REVIEW

**Comment by Rocss** [comment]
Currently this is being tested by Express g18n team. Will come back with updating all the necessary documentation when it is validated.

**Comment by Rocss** [comment]
I am marking this as ready for review as it was finally tested by Express g18n team and confirmed that number field works well for Hindi and Bengali keyboards.
Happy to address any other code review, I don't know what I missed 😅 


@majornista I think this PR is affecting your PR: https://github.com/adobe/react-spectrum/pull/6520, would you like me to add additional checks to mine?

File: packages/react-aria-components/docs/NumberField.mdx [hunk 1]
```diff
@@ -128,7 +128,7 @@ the stepper buttons. `NumberField` helps achieve accessible number fields that s
 formatting options and can be styled as needed.
 
 * **Customizable formatting** – Support for internationalized number formatting and parsing including decimals, percentages, currency values, and units. Multiple currency formats are supported, including symbol, code, and name in standard or accounting notation.
-* **International** – Support for the Latin, Arabic, and Han positional decimal numbering systems in [over 30 locales](internationalization.html#supported-locales). The numbering system is automatically detected from user input, and input method editors such as Pinyin are supported.
+* **International** – Support for the Latin, Arabic, and Han, Devanagari and Bengali positional decimal numbering systems in [over 30 locales](internationalization.html#supported-locales). The numbering system is automatically detected from user input, and input method editors such as Pinyin are supported.
```
Comment (snowystinger) [inline packages/react-aria-components/docs/NumberField.mdx:131]:
```suggestion
* **International** – Support for the Latin, Arabic, Devanagari, Bengali, and Han positional decimal numbering systems in [over 30 locales](internationalization.html#supported-locales). The numbering system is automatically detected from user input, and input method editors such as Pinyin are supported.
```

from what I can tell, Devanagari and Bengali are already positional decimal numbering systems, where as Han is not necessarily, which is why we clarify that we support that particular style of it

**Review by snowystinger — COMMENTED** [review]
looks pretty good, thanks for the tests.

**Review by snowystinger — APPROVED** [review]

**Comment by Rocss** [comment]
I no longer have the option to add additional reviewers, @snowystinger would you help me ping additional RS members for code review?

**Review by LFDanLu — APPROVED** [review]
LGTM, verified I was able to type various numeric values via IDE into the NumberField when Devanagari and Bengali was enabled

**Pull request merged**